src/sql-parser/src/parser.rs

```diff
@@ -5341,6 +5341,16 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parse `UP TO`, if present
+    fn parse_optional_up_to(&mut self) -> Result<Option<Expr<Raw>>, ParserError> {
```

Comment:
Can you add some test cases to `test/testdata` in this create? It looks like we don't have any existing SUBSCRIBE tests which is probably an oversight. You don't the need to add comprehensive tests for all SUBSCRIBE parsing, just this one feature.